src/common/operator/string_cast.cpp

```diff
@@ -159,6 +159,56 @@ duckdb::string_t StringCast::Operation(timestamp_t input, Vector &vector) {
 	return result;
 }
 
+template <>
+duckdb::string_t StringCast::Operation(timestamp_ns_t input, Vector &vector) {
+	if (input == timestamp_t::infinity()) {
+		return StringVector::AddString(vector, Date::PINF);
+	} else if (input == timestamp_t::ninfinity()) {
+		return StringVector::AddString(vector, Date::NINF);
+	}
+
+	date_t date_entry;
+	dtime_t time_entry;
+	int32_t nanos;
+	Timestamp::Convert(input, date_entry, time_entry, nanos);
```

Comment:
This seems very similar to the timestamp cast above - could we try to unify them, e.g. with `HAS_NANOS` as a template parameter to the same underlying function?